manticore/core/smtlib/expression.py

```diff
@@ -159,126 +260,101 @@ def __rxor__(self, other):
         return BoolXor(self.cast(other), self)
 
     def __bool__(self):
-        # try to be forgiving. Allow user to use Bool in an IF sometimes
-        from .visitors import simplify
-
-        x = simplify(self)
-        if isinstance(x, Constant):
-            return x.value
-        raise NotImplementedError("__bool__ for Bool")
-
-
-class BoolVariable(Bool):
-    __slots__ = ["_name"]
-
-    def __init__(self, name: str, *args, **kwargs):
-        assert " " not in name
-        super().__init__(*args, **kwargs)
-        self._name = name
-
-    @property
-    def name(self):
-        return self._name
-
-    def __copy__(self, memo=""):
-        raise ExpressionException("Copying of Variables is not allowed.")
-
-    def __deepcopy__(self, memo=""):
-        raise ExpressionException("Copying of Variables is not allowed.")
-
-    def __repr__(self):
-        return "<{:s}({:s}) at {:x}>".format(type(self).__name__, self.name, id(self))
-
-    @property
-    def declaration(self):
-        return f"(declare-fun {self.name} () Bool)"
-
-
-class BoolConstant(Bool):
-    __slots__ = ["_value"]
+        raise ExpressionError(
+            "You tried to use a Bool Expression as a boolean constant. Expressions could represent a set of concrete values."
+        )
```

Comment:
This statement is causing issues. I've replaced it with just 
```suggestion
        return self
```
and it seems to be passing for me natively on both Python 3.6 and 3.9 🤷 

Maybe we should revert this to how it was.